lib/download-file.ts

```diff
@@ -72,7 +72,7 @@ async function streamWithProgress(
       writer.write(Buffer.from(chunk));
       if (progressCallback != null) {
         bytesDone += chunk.byteLength;
-        const percent: number = length === 0 ? null : Math.floor(bytesDone / length * 100);
+        const percent: number | undefined = length === 0 ? undefined : Math.floor(bytesDone / length * 100);
```

Comment:
`percent` gets passed into ByteProgressCallback's optional `percent` parameter, `undefined` matches it